src/main/java/org/elasticsearch/search/aggregations/Aggregator.java

```diff
@@ -16,36 +16,54 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.elasticsearch.search.aggregations;
 
-import com.google.common.base.Predicate;
-import com.google.common.collect.Iterables;
+package org.elasticsearch.search.aggregations;
 
-import org.apache.lucene.index.LeafReaderContext;
-import org.apache.lucene.search.Scorer;
 import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.lease.Releasable;
-import org.elasticsearch.common.util.BigArrays;
 import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.search.aggregations.bucket.BucketsAggregator;
-import org.elasticsearch.search.aggregations.bucket.DeferringBucketCollector;
 import org.elasticsearch.search.aggregations.support.AggregationContext;
 import org.elasticsearch.search.internal.SearchContext;
-import org.elasticsearch.search.internal.SearchContext.Lifetime;
-import org.elasticsearch.search.query.QueryPhaseExecutionException;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.EnumSet;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
 
+/**
+ * An Aggregator.
+ */
+// IMPORTANT: DO NOT add methods to this class unless strictly required.
+// On the other hand, if you can remove methods from it, you are highly welcome!
 public abstract class Aggregator extends BucketCollector implements Releasable {
```

Comment:
With the exception of the `descendsFromBucketAggregator` method, which could probably be moved to AggregatorBase as it's an implementation detail and not integral to the idea of an Aggregator, all the methods in this class are abstract. Should we make Aggregator an interface?